File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3084_count-substrings-starting-and-ending-with-given-character/solutions/cpp/solution.cpp
#include <vector>
#include <string>

class Solution {
public:
    long long countSubstrings(std::string s, char c) {
        std::vector<int> positions;
        long long count = 0;

        for (int i = 0; i < s.length(); ++i) {
            if (s[i] == c) {
                positions.push_back(i);
            }
        }

        for (int index : positions) {
            count += (index + 1) * (s.length() - index);
        }

        return count;
    }
};

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3084_count-substrings-starting-and-ending-with-given-character/solutions/java/solution.java
class Solution {
    public long countSubstrings(String s, char c) {
        long count = 0;
        long totalSubstrings = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == c) {
                count++;
            }
        }
        totalSubstrings = count * (count + 1) / 2;
        return totalSubstrings;
    }
}
